src/MessageTrait.php

```diff
@@ -105,35 +89,12 @@ public function withHeader($header, $value): self
 
     public function withAddedHeader($header, $value): self
     {
-        if (!is_array($value)) {
-            $value = [$value];
-        } elseif (!empty($value)) {
-            $value = array_values($value);
-        } else {
-            throw new \InvalidArgumentException('Header values must be strings');
-        }
-
         if (!is_string($header) || empty($header)) {
```

Comment:
Why do we still have validation here when this should be handled by `validateAndTrimHeader`?

Reply:
True. It should be removed.

Reply:
No, If $header is not a string (ie an object) it will fail when we try to use it as an array key.